libdevice/fallback-cstring.cpp

```diff
@@ -7,9 +7,49 @@
 //===----------------------------------------------------------------------===//
 
 #include "wrapper.h"
+#include <cstdint>
 
 #ifdef __SPIR__
+DEVICE_EXTERN_C
 void *__devicelib_memcpy(void *dest, const void *src, size_t n) {
   return __builtin_memcpy(dest, src, n);
 }
+
+DEVICE_EXTERN_C
+void *__devicelib_memset(void *dest, int c, size_t n) {
+  return __builtin_memset(dest, c, n);
+}
+
+DEVICE_EXTERN_C
+int __devicelib_memcmp(const void *s1, const void *s2, size_t n) {
+  if (s1 == s2 || n == 0)
+    return 0;
+
+  const uint32_t *s1_uint32_ptr = reinterpret_cast<const uint32_t *>(s1);
```

Comment:
that's dangerous. What if s1 or s2 is not 4-byte aligned?

Reply:
Hi, @kbobrovs 
Thanks for pointing out this issue, I will update the patch to handle 4-byte alignment first.
Thanks very much.

Reply:
Hi, @kbobrovs 
I have updated the patch, if s1 or s2 is not 4 byte aligned, we will use a simple 1byte cmp loop to check.
Thanks very much.

Reply:
Thanks. This LGTM now.